tls/s2n_tls13_secrets.c

```diff
@@ -671,3 +702,51 @@ S2N_RESULT s2n_tls13_secrets_get(struct s2n_connection *conn, s2n_extract_secret
     RESULT_ENSURE_GT(secret->size, 0);
     return S2N_RESULT_OK;
 }
+
+int s2n_connection_tls_exporter(struct s2n_connection *conn,
+        const uint8_t *label_in, uint32_t label_length,
+        const uint8_t *context, uint32_t context_length,
+        uint8_t *output_in, uint32_t output_length)
+{
+    POSIX_ENSURE_REF(conn);
+    POSIX_ENSURE_REF(output_in);
+    POSIX_ENSURE_REF(label_in);
+    POSIX_ENSURE_REF(context);
+    POSIX_ENSURE(s2n_connection_get_protocol_version(conn) == S2N_TLS13, S2N_ERR_INVALID_STATE);
+
+    POSIX_ENSURE(is_handshake_complete(conn), S2N_ERR_HANDSHAKE_NOT_COMPLETE);
+    POSIX_ENSURE_REF(conn->secure);
+    POSIX_ENSURE_REF(conn->secure->cipher_suite);
+    s2n_hmac_algorithm hmac_alg = conn->secure->cipher_suite->prf_alg;
+
+    struct s2n_blob label = { 0 };
+    POSIX_GUARD(s2n_blob_init(&label, (uint8_t *) label_in, label_length));
```

Comment:
This cast is breaking the clang lint since it's casting away `const`. I'm not sure what the typical workaround for this is - the `label` blob isn't written to, but it seems like there's not an initializer that ensures that? Maybe we should be allocating (or creating a ~249 byte stack buffer) and copying into it?

Reply:
Yeah unfortunately we don't have a great solution for that problem. A copy is never really ideal, but in practice it should be pretty cheap for only 250ish bytes, and that would be the safest option to ensure the "const" guarantee. Or you can add a comment explaining why this is safe and force the conversion using [uintptr_t](https://github.com/aws/s2n-tls/blob/ab7bfce9b78109c26600d1bb63c555643be1c345/tls/s2n_send.c#L251), but that always risks us violating the "const" promise.